src/app/news/news-list.component.ts

```diff
@@ -88,20 +92,32 @@ export class NewsListComponent implements OnInit, OnChanges, AfterViewInit, OnDe
   }
 
   ngAfterViewInit() {
+    this.setupObserver();
+  }
+
+  setupObserver() {
```

Comment:
This breaks up the standard Angular lifecycle methods. I understand there's a benefit to having this next to the `AfterViewInit` that invokes it, but I missed it on my first readthrough because I didn't expect a custom method before the `OnDestroy`.